### PR TITLE
ASA 8392

### PR DESCRIPTION
**Pull request by vishalhcl-5960**
https://jira02.hclpnp.com/browse/ASA-8392 - As a DevOps Engineer, I want to automate DAST Scan execution in AppScan 360 via HCL AppScan Jenkins Plugin.

### CODE REVIEW

File: src/main/resources/com/hcl/appscan/jenkins/plugin/Messages.properties [hunk 1]
```diff
@@ -22,8 +22,7 @@ error.threshold.invalid=Invalid entry. Enter an integer value.
 error.malformed.url=Malformed URL {0}.
 error.server.unavailable=AppScan plugin encountered an unexpected error while connecting to AppScan on Cloud.
 error.mobile.analyzer=Incorrect user input. The legacy Mobile Analyzer technology is no longer supported, we recommend using our Static Analyzer scanning for Mobile Applications.
-error.dynamic.AppScan360= Dynamic Analysis is available for AppScan on Cloud only.
-error.dynamic.analyzer.AppScan360= HCL AppScan task failed: Dynamic analysis (DAST) is available in AppScan on Cloud only.
+error.presence.AppScan360= Presence is available for AppScan on Cloud only.
```
Comment (kripajoym) [inline src/main/resources/com/hcl/appscan/jenkins/plugin/Messages.properties:25]: in or For ?

File: src/main/java/com/hcl/appscan/jenkins/plugin/auth/JenkinsAuthenticationProvider.java [hunk 2]
```diff
@@ -102,9 +102,10 @@ private String getToken() {
 		return Secret.toString(m_credentials.getToken());
 	}
 
-	public boolean isAppScan360(){
-		String url = m_credentials.getUrl();
-        	return  url != null && !url.isEmpty() && !url.contains("appscan.com");
+	@Override
```
Comment (kripajoym) [inline src/main/java/com/hcl/appscan/jenkins/plugin/auth/JenkinsAuthenticationProvider.java:105]: Is this required to be override ? Can we add this is only in Jenkins Auth.

File: src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java [hunk 3]
```diff
@@ -9,6 +9,7 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.Serializable;
+import java.net.Proxy;
```
Comment (kripajoym) [inline src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java:12]: why is this import required? there is no changes related to Proxy

File: src/main/java/com/hcl/appscan/jenkins/plugin/scanners/DynamicAnalyzer.java [hunk 4]
```diff
@@ -325,17 +326,22 @@ public FormValidation doCheckScanFile(@QueryParameter String scanFile) {
 			return FormValidation.ok();
 		}
 
-		public FormValidation doCheckTarget(@QueryParameter String target,@RelativePath("..") @QueryParameter String credentials, @AncestorInPath ItemGroup<?> context, @QueryParameter String presenceId) {
+		public FormValidation doCheckTarget(@QueryParameter String target,@RelativePath("..") @QueryParameter String credentials, @AncestorInPath ItemGroup<?> context, @QueryParameter String presenceId) throws IOException {
```
Comment (kripajoym) [inline src/main/java/com/hcl/appscan/jenkins/plugin/scanners/DynamicAnalyzer.java:329]: None of the Formvalidations are throwing IO Exception why is it required ?

File: src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java (hunk 3 above)
Comment (vishalhcl-5960) [reply src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java:12]: Removed this import.

File: src/main/java/com/hcl/appscan/jenkins/plugin/scanners/DynamicAnalyzer.java (hunk 4 above)
Comment (vishalhcl-5960) [reply src/main/java/com/hcl/appscan/jenkins/plugin/scanners/DynamicAnalyzer.java:329]: Removed the IO exception.

File: src/main/java/com/hcl/appscan/jenkins/plugin/auth/JenkinsAuthenticationProvider.java (hunk 2 above)
Comment (vishalhcl-5960) [reply src/main/java/com/hcl/appscan/jenkins/plugin/auth/JenkinsAuthenticationProvider.java:105]: As we are now using the isAppScan360() method in the appscan-plugin instead of the SDK. I will not go with overriding this method.

File: src/main/resources/com/hcl/appscan/jenkins/plugin/Messages.properties (hunk 1 above)
Comment (vishalhcl-5960) [reply src/main/resources/com/hcl/appscan/jenkins/plugin/Messages.properties]: It should be "in".
